### PR TITLE
[Cherry-pick release/2.14] Fix batchnorm accuracy by sycl::rsqrt

### PR DESCRIPTION
**Pull request by torchxpubot**
Align  rsqrt with inductor. Should fix timm_efficientnet accuracy issue in https://github.com/intel/torch-xpu-ops/issues/4905. For detectron2_maskrcnn and detectron2_maskrcnn_r_50_fpn, add them to use_iou_for_bool_masks
After https://github.com/pytorch/pytorch/pull/190206, inductor use rsqrt instead of 1/sqrt
Currently, the compute order of eager and inductor is still different, but changing them will cause UT accuracy fail

Cherry-pick of #4973 to `release/2.14`.
Cherry-picked from commit 38ba1c75eb0863d776dbc5ec60d349fc0563c7ee.

### CODE REVIEW

**Review by jianyizh — APPROVED** [review]

**Comment by chuanqi129** [comment]
@torchxpubot check-ut
